# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: d01a1a055fec
# Full Hash: d01a1a055fece12b906b938954f984bd69278ef7
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-04-30 20:49:53
# Regressor Bug: 1960824
# File Overlap Count: 1
# Description:
#   Bug 1960824 - Introduce nsICookie::SAMESITE_UNSET, r=edgul,webdriver-reviewers,extension-reviewers,cookie-reviewers,valentin,Sasha,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D246638
# ==============================================================================

diff -r 307ff391dd02 -r d01a1a055fec dom/base/Document.cpp
--- a/dom/base/Document.cpp	Wed Apr 30 08:53:12 2025 +0000
+++ b/dom/base/Document.cpp	Wed Apr 30 08:59:18 2025 +0000
@@ -6781,9 +6781,11 @@
         continue;
       }
 
+      nsCOMPtr<nsIURI> cookieURI = cookiePrincipal->GetURI();
+
       if (thirdParty &&
           !CookieCommons::ShouldIncludeCrossSiteCookie(
-              cookie, CookieJarSettings()->GetPartitionForeign(),
+              cookie, cookieURI, CookieJarSettings()->GetPartitionForeign(),
               IsInPrivateBrowsing(), UsingStorageAccess(), on3pcbException)) {
         continue;
       }
@@ -6925,7 +6927,7 @@
 
   if (thirdParty &&
       !CookieCommons::ShouldIncludeCrossSiteCookie(
-          cookie, CookieJarSettings()->GetPartitionForeign(),
+          cookie, documentURI, CookieJarSettings()->GetPartitionForeign(),
           IsInPrivateBrowsing(), UsingStorageAccess(), on3pcbException)) {
     return;
   }